dace/frontend/fortran/intrinsics.py

```diff
@@ -1025,22 +1586,30 @@ def generate_scale(arg: ast_internal_classes.Call_Expr_Node):
         const_two = ast_internal_classes.Int_Literal_Node(value="2")
 
         # I and RADIX(X) are both integers
-        rval = ast_internal_classes.Call_Expr_Node(name=ast_internal_classes.Name_Node(name="pow"),
-                                                   type="INTEGER",
-                                                   args=[const_two, i],
-                                                   line_number=line)
+        rval = ast_internal_classes.Call_Expr_Node(
+            name=ast_internal_classes.Name_Node(name="pow"),
+            type="INTEGER",
+            args=[const_two, i],
+            line_number=line,
+            subroutine=False,
+        )
 
         mult = ast_internal_classes.BinOp_Node(op="*", lval=x, rval=rval, line_number=line)
 
         # pack it into parentheses, just to be sure
         return ast_internal_classes.Parenthesis_Expr_Node(expr=mult)
 
+    def generate_epsilon(arg: ast_internal_classes.Call_Expr_Node):
+        ret_val = sys.float_info.epsilon
```

Comment:
Epsilon value should depend on float type (if you use this, REAL4 will underflow)